### PR TITLE
fix: remove extra arguments from wrangler init deprecation message and update recommended c3 version in message

### LINKED ISSUE #4480: 🚀 Feature Request: Update wrangler init deprecation warning
### Describe the solution

The current warning reads:

`Please use "npm create cloudflare@2 my-script -- --type pre-existing --existing-script my-script" instead.`

c3 can infer the `pre-existing` type from the presence of the `--existing-script` flag, so we can shorten this to:

`Please use "npm create cloudflare@2 my-script -- --existing-script my-script" instead.`

I'd also suggest bumping the version to `2.5.0`. This version introduces an auto-update feature that will make sure users get the latest minor version of c3 and prevent problems where older 2.x.x versions get cached by previous runs `wrangler init`.

### PR DESCRIPTION
**Pull request by admah**
Fixes #4480 .

**What this PR solves / how to test:**

This PR updates the deprecation message presented to users that run `wrangler init --from-dash` by removing an extra `type` argument and referencing a more recent version of C3 that auto-updates the minor version preventing caching issues users faced if they had previously run `wrangler init`

**Author has addressed the following:**

- Tests
  - [x] Included: updated snapshots
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] Included
  - [ ] Not necessary because:
- Associated docs
  - [ ] Issue(s)/PR(s):
  - [x] Not necessary because:

**Note for PR author:**

We want to celebrate and highlight awesome PR review! If you think this PR received a particularly high-caliber review, please assign it the label `highlight pr review` so future reviewers can take inspiration and learn from it.

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Looks like legitimate fails in CI.

**Comment by admah** [comment]
@petebacondarwin it looks like failures in unrelated parts of the codebase. I tried running again, but it failed. Is there something I'm not seeing?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Need to fix the CI jobs before we can land this.

**Comment by petebacondarwin** [comment]
> I understand wanting to ensure users run a version of C3 which checks for updates which, I assume, was a feature introduced in 2.5.0 but it's a bit redundant to give users a command which downloads 2.5.0 to only then download the latest version. It should probably be whatever is the latest version right now.
> 
> We can also read that value from the C3 package.json so it's somewhat up-to-date with each release but I would need to look into whether that would be the version before each release as the package.json _might_ be updated during the release process _after_ the builds

What's the action regarding this PR here @RamIdeas? Are you proposing a change to this PR? I feel like your concern is separate from this specific PR and perhaps could be dealt with in a follow up?

**Review by petebacondarwin — APPROVED** [review]
CI is now green. LGTM.
@RamIdeas - please mark as "request changes" if there is anything blocking merging this.

**Comment by admah** [comment]
@jculvey do you have any thoughts on the comment above? https://github.com/cloudflare/workers-sdk/pull/4833#issuecomment-1932149769

This PR intends to address #4480

**Comment by RamIdeas** [comment]
My comments above are not blocking. We can follow up later if the double download is affecting a critical number of users

**Review by RamIdeas — APPROVED** [review]
